### PR TITLE
Replace CODEOWNERS with CODENOTIFY

### PR DESCRIPTION
**Pull request by nicksnyder**
# Background

This PR replaces https://github.com/sourcegraph/sourcegraph/pull/11718.

For reasons mentioned in that other PR, I suspect that we would be better off not using CODEOWNERS.

The anonymous survey indicates that the dev team agrees:
Screen Shot 2020-09-15 at 8 42 15 PM
Source: https://docs.google.com/forms/d/e/1FAIpQLSfjDJxP1wdfEuTZdeNqTsCxfXD2HIYx_5aiTC17B6LyXwYorw/viewanalytics

The biggest concern was devs not being notified of changes that they care about.

# This PR

This PR is the beginning of two related, but independent, experiments:
1. We are not going to use CODEOWNERS for the rest of September, and October.
2. I have created a tool called [codenotify](https://github.com/sourcegraph/codenotify) which aims to address the use case of allowing devs to be notified of changes that they care about (without creating the expectation of review). After this is merged, devs can experiment with adding CODENOTIFY files. Feedback on the behavior of codenotify is obviously welcome at any time.

At the end of October I will send out a survey to see how the team feels after living with these changes for a full release cycle.

Valid outcomes of this experiment are:

1. Everyone hates this and we go back to using only CODEOWNERS.
1. Devs find value in codenotify but want to use CODEOWNERS as well in some cases.
1. Devs are happy to only use codenotify and keep CODEOWNERS empty.

Given the data already collected in the previous survey, and my conviction that this is worth trying, I plan on merging this tomorrow.

### CODE REVIEW

**Comment by bobheadxi** [comment]
A few super minor nits about codenotify:

* could the bot not post a comment if there are no notifications? (ie in the comment above)
* CODENOFITY all caps is a bit of an aggressive header 😅 
* A footer with a link to how to subscribe to notifications might be helpful, eg:

<sub>Want notifications for changes to these files? <a href="https://github.com/sourcegraph/codenotify#codenotify-files">Add a CODENOTIFY file</a>.</sub>

**Comment by abitrolly** [comment]
> For reasons mentioned in that other PR

A short summary wouldn't help for those with no idea.

**Comment by nicksnyder** [comment]
Thanks for the feedback @bobheadxi! I addressed it here: https://github.com/sourcegraph/codenotify/compare/v0.1...v0.2

You can see what a rendered report looks like in the updated readme: https://github.com/sourcegraph/codenotify#github-action

**Review by beyang — APPROVED** [review]

File: .github/workflows/codenotify.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+on: [pull_request]
```
Comment (bobheadxi) [inline .github/workflows/codenotify.yml:1]:
```suggestion
name: Codenotify
on: [pull_request]
```

**Review by tsenart — APPROVED** [review]
Exciting experiments!

**Pull request merged**